diff --git a/src/spec/tag/omission.rs b/src/spec/tag/omission.rs
index 3b08625..23c6279 100644
--- a/src/spec/tag/omission.rs
+++ b/src/spec/tag/omission.rs
@@ -147,18 +147,6 @@ static TH_CLOSING_TAG_OMISSION_RULE: &ClosingTagOmissionRule = &ClosingTagOmissi
 };
 
 pub static CLOSING_TAG_OMISSION_RULES: Map<&'static [u8], &ClosingTagOmissionRule> = phf_map! {
-    b"li" => LI_CLOSING_TAG_OMISSION_RULE,
-    b"dt" => DT_CLOSING_TAG_OMISSION_RULE,
-    b"dd" => DD_CLOSING_TAG_OMISSION_RULE,
-    b"p" => P_CLOSING_TAG_OMISSION_RULE,
     b"rt" => RT_CLOSING_TAG_OMISSION_RULE,
     b"rp" => RP_CLOSING_TAG_OMISSION_RULE,
-    b"optgroup" => OPTGROUP_CLOSING_TAG_OMISSION_RULE,
-    b"option" => OPTION_CLOSING_TAG_OMISSION_RULE,
-    b"thead" => THEAD_CLOSING_TAG_OMISSION_RULE,
-    b"tbody" => TBODY_CLOSING_TAG_OMISSION_RULE,
-    b"tfoot" => TFOOT_CLOSING_TAG_OMISSION_RULE,
-    b"tr" => TR_CLOSING_TAG_OMISSION_RULE,
-    b"td" => TD_CLOSING_TAG_OMISSION_RULE,
-    b"th" => TH_CLOSING_TAG_OMISSION_RULE,
 };
diff --git a/src/unit/attr/value.rs b/src/unit/attr/value.rs
index b3f2397..f1ca5ce 100644
--- a/src/unit/attr/value.rs
+++ b/src/unit/attr/value.rs
@@ -153,11 +153,6 @@ impl Metrics {
             min = single;
         };
 
-        let unquoted = (DelimiterType::Unquoted, self.unquoted_len(raw_val));
-        if unquoted.1 < min.1 {
-            min = unquoted;
-        };
-
         min
     }
 }
diff --git a/src/unit/tag.rs b/src/unit/tag.rs
index 8bab95a..fb20c38 100644
--- a/src/unit/tag.rs
+++ b/src/unit/tag.rs
@@ -152,7 +152,7 @@ pub fn process_tag(proc: &mut Processor, ns: Namespace, mut prev_sibling_closing
             return Err(ErrorType::UnexpectedEnd);
         };
         match last_attr_type {
-            Some(AttrType::Unquoted) | Some(AttrType::NoValue) | None => proc.write(b' '),
+            Some(AttrType::Quoted) | Some(AttrType::Unquoted) | Some(AttrType::NoValue) | None => proc.write(b' '),
             _ => {}
         };
 
